packages/amplify-cli/src/extensions/amplify-helpers/on-category-outputs-change.ts

```diff
@@ -63,3 +66,25 @@ function attachContextExtensions(context, packageLocation) {
     }
   }
 }
+
+function migrateAmplifyMeta(context, amplifyMeta?) {
```

Comment:
We need a more specific name here and some comments on what it migrated FROM and TO, feel free to link issue # as well.

This only invoked when a category is changing, what happens to customers who are not running a `push` or `update auth`, just doing a `pull` over the existing, already initialized project with the new CLI that has these changes?

If we have a migration of state files in a PR we have to add migration tests to cover and verify that everything works correctly.